app/containers/GamePage/index.js

```diff
@@ -75,13 +94,46 @@ export default class GamePage extends Component {
     });
   }
 
+  loadParticipants() {
+    const { database, participants } = this.state;
+    const { params } = this.props;
+    const usersRef = database.ref(`${params.rid}/participants`);
+    usersRef.off();
+    usersRef.limitToLast(12).on('child_added', (data) => {
+      const val = data.val();
+      participants.push({ key: data.key, name: val.name, role: '', health: 5, hand: [] });
+      this.setState({ participants });
+      usersRef.once('value', (snapshot) => {
```

Comment:
Why do you need to even check the snapshot.numChidren() when this num can be taken from participants.length. Get rid of this .once listener and get rid of players from the state. You can just use participants.length to get the length.